server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -319,6 +311,14 @@ public HttpBody.Stream contentStream() {
         return httpRequest.body().asStream();
     }
 
+    /**
+     * Release underlying HTTP request and related buffers.
+     */
+    @Override
+    public void close() {
```

Comment:
Is this _only_ for the `testPipelineOverflow` workaround? If so, I'd rather we used a package-private method instead of making this public and `Releasable`. The eventual goal would be to release these buffers up-front and then we can remove the workaround, and in the meantime we don't want to accumulate other spots that explicitly close the request.

Reply:
I can add method that returns HttpBody and then we can close body explicitly. Right now there is public access from RestRequest to close pooled/streamed content.